draft-gruessing-moq-requirements.md

```diff
@@ -265,35 +265,25 @@ Packaging of media describes how encapsulation of media to carry the raw media w
 * Within the protocol itself, where the protocol defines the carrying for each media encoding the ancillary data required for decoding the media.
 * A common encapsulation format such as ISOBMFF which defines a generic method for all media and handles ancillary decode information.
 
-The working group must agree on which approach should be taken to the packaging of media, taking into consideration the various technical trade offs that each provide.
-
-### Use existing container format like CMAF
-
-### Support Raw container format
-
-### Do we need other formats ?
+The working group must agree on which approach should be taken to the packaging of media, taking into consideration the various technical trade offs that each provide. If the working group decides on a common encapsulation format, the mechanisms within the protocol SHOULD allow for new encapsulation formats to be used.
 
 ## Media Consumption {#med-consumption}
 
-### Consumers are authorized at the Origin.
-
-### Allows consumers to ask for the media data by name.
-
-### Allows consumers to ask the right quality/variant.
+Receivers SHOULD be able to as part of negotiation of a session {{MOQ-negotiation}} specify which media to receive, not just with respect to the media format and codec, but also the varient thereof such as resolution or bitrate.
 
 ## Relays, Caches, and other MOQ Network Elements {#MOQ-network-entities}
 
-### Allow caching and distribution of media data as applicable
+### Pull & Push
 
-### Store/distribute the data via trust relayed by Origin
+To enable use cases where receivers may wish to address a particular time of media in addition to having the most recently produced media available, both "pull" and "push" of media SHOULD be supported, with consideration that producers and intermediates SHOULD also signal what media is available (commonly referred to as a "DVR window"). Behaviours around cache durations for each MoQ entity should be defined.
```

Comment:
We have had meaningful conversations about push operation, but I'm not sure that the group agrees that both push and pull operation should be supported. I might be misidentifying the speaker, but I thought @kixilated had talked about using long pulls to simulate push operation. 

So, no objection to including this point, but I'm recognize we may have another conversation about this soon.

Reply:
When I talk about pull, I mean two possibilities:
* A receiver deliberately wanting media not on live-edge (playing media from a different offset in the DVR window)
* A receiver wanting a re-send of data, such as non-time sensitive data, or previous frames to assist in current GOP decoding

If you think this might muddy the discussion, I'm happy to remove it and turn into an issue for later.

Reply:
I think this is pushing on the conversation I was having with Will Law, about the difference between very small buffers used for hop-by-hop reliability and larger buffers that are closer to being parts of caches. I'm reading your second bullet as closer to the first category ("small buffers") - am I reading that correctly?

I don't think removing this idea will make the discussion less muddy, though. :grimacing: You should probably leave it for now.

Reply:
Okay, I'll leave it for now.